### PR TITLE
create osd, init and run in one shot, without the need to run ceph os…

### PR DESCRIPTION
**Pull request by streamnsight**
this changes the logic on osd_directory endpoint

instead of having to run `ceph osd create` on then monitor, then create a directory manually on the host, chown it etc...  before running the daemon, this is done in the entrypoint to start an OSD in osd_directory mode

the original logic was looking for osd ids in the `/var/lib/ceph/osd/` folder however it doesn't make sense to have multiple OSD directories on the same host/drive as each will report full disk size as their own.
Here we automate creation of the OSD (no more manual steps) and prevent creation of multiple osd_directory at the same time.

this may be a breaking change if anyone was actually running multiple osd_directories on the same host, but my tests show it doesn't work well, so it must not be used that way.

PS: this PR goes along with the other PRs to populate KV store and bug fixes. it does not work well without those fixes.

### CODE REVIEW

File: daemon/entrypoint.sh [hunk 1]
```diff
@@ -188,12 +188,26 @@ function start_osd {
...
-    echo "ERROR- could not find any OSD, did you bind mount the OSD data directory?"
-    echo "ERROR- use -v <host_osd_data_dir>:<container_osd_data_dir>"
+  if [[ ! -d /var/lib/ceph/osd ]]; then
+    echo "ERROR- could not find the osd directory, did you bind mount the OSD data directory?"
+    echo "ERROR- use -v <host_osd_data_dir>:/var/lib/ceph/osd"
     exit 1
   fi
 
+  # make sure ceph owns the directory
+  chown ceph. /var/lib/ceph/osd
```
Comment (Ulexus) [inline daemon/entrypoint.sh]: This should probably be a recursive chown, and I don't see a reason not to set the group, as well (though perhaps that is too invasive):  `chown -R ceph:ceph /var/lib/ceph/osd`

File: daemon/entrypoint.sh [hunk 2]
```diff
@@ -188,12 +188,26 @@ function start_osd {
...
+
+  # check if anything is there, if not create an osd with directory
+  if [[ -n "$(find /var/lib/ceph/osd -prune -empty)" ]]; then
+    echo "Creating osd with ceph osd create"
+    OSD_ID=$(ceph osd create)
```
Comment (Ulexus) [inline daemon/entrypoint.sh]: What happens here if the bootstrap or admin key isn't found?  We probably need a failure check here.

File: daemon/entrypoint.sh (hunk 1 above)
Comment (leseb) [inline daemon/entrypoint.sh]: actually `ceph.` = `ceph:ceph`

Comment (Ulexus) [inline daemon/entrypoint.sh]: Interesting;  I've never seen that abbreviation... I see it mentioned in the man page, though.  It's smarter than that, even; it implies the primary group of that user's login, not necessarily the same name.  Nice.

Comment (leseb) [inline daemon/entrypoint.sh]: :)

File: daemon/entrypoint.sh (hunk 2 above)
Comment (streamnsight) [inline daemon/entrypoint.sh]: added the below to catch if OSD_ID is not a number (and some other error was returned)

**Comment by Ulexus** [comment]
lgtm

**Comment by leseb** [comment]
LGTM

**Pull request merged**